content/software-testing-the-hardware-software-interface-part-2.md

```diff
@@ -0,0 +1,417 @@
+---
+title: "Embedded Software Testing: Practical Continuous Integration with Hardware in the Loop (Part 2)"
+date: 2024-11-03
+author: Leonardo Held, Torizon Developer
+draft: false
+
+tags: ["Testing", "Embedded", "Linux", "Embedded"]
+
+abstract: A series about Embedded Software Testing.
+
+image: /soave-gate.jpg
+---
+
+This is the second installment in the embedded software testing series. This one will be about the architecture of a 
+system to test embedded software. We'll briefly cover the what is usually called a "dual-targetting" approach as well.
+
+### Theory
+
+In the previous installment we talked about the different levels of software testing and the last one was "end-to-end
+testing", which is treating the hardware or software Under Test (UDT) like a black box, poking it with a certain stimuli
```

Comment:
UDT? DUT is ok for hardware/device and for software is normally use SUT (system under test/software under test as you wish)

Reply:
I've changed to use DUT as I do in the diagram below. Thanks.